test/e2e/framework/job/wait.go

```diff
@@ -99,22 +101,30 @@ func WaitForJobGone(c clientset.Interface, ns, jobName string, timeout time.Dura
 	})
 }
 
-// CheckForAllJobPodsRunning uses c to check in the Job named jobName in ns is running. If the returned error is not
-// nil the returned bool is true if the Job is running.
-func CheckForAllJobPodsRunning(c clientset.Interface, ns, jobName string, parallelism int32) (bool, error) {
+// EnsureJobPodsRunning uses c to check in the Job named jobName in ns
+// is running, returning an error if the expected parallelism is not
+// satisfied.
+func EnsureJobPodsRunning(c clientset.Interface, ns, jobName string, parallelism int32) error {
 	label := labels.SelectorFromSet(labels.Set(map[string]string{JobSelectorKey: jobName}))
 	options := metav1.ListOptions{LabelSelector: label.String()}
 	pods, err := c.CoreV1().Pods(ns).List(options)
 	if err != nil {
-		return false, err
+		return err
 	}
+	pods := make([]string, 0, parallelism)
 	count := int32(0)
 	for _, p := range pods.Items {
 		if p.Status.Phase == v1.PodRunning {
 			count++
+			pods = append(pods, fmt.Sprintf("%s (%s)", p.ObjectMeta.Name, p.Status.Phase))
```

Comment:
Why not just always add and print all that?

Reply:
> Why not just always add and print all that?

Because I don't really care what the`Message` is if the pod is running.  But consolidated to a single `append` in 3e621c7 -> 96b04bf, squashing your wking#1. Thanks :).